cdap-ui/server/express.js

```diff
@@ -471,6 +449,44 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
     },
   ]);
 
+  app.all('/api/:path(*)', function(req, res) {
+    var opts = {
+      method: req.method,
+      url: urlhelper.constructUrl(cdapConfig, req.param('path')),
+      headers: {},
+    };
+
+    if (req.headers['authorization']) {
+      opts.headers['Authorization'] = req.headers['authorization'];
+    }
+    if (req.headers['content-type']) {
+      opts.headers['Content-Type'] = req.headers['content-type'];
+    }
+
+    if (['PUT', 'POST'].indexOf(req.method) !== -1) {
+      opts.body = req.body;
+      opts.json = true;
```

Comment:
It may not be json. Why need to hardcode this?

Reply:
Can you give an example where this will not be JSON? We used to have streams which used to just pass in string. But we don't seem to be having that anymore. It would be hard to determine if this is different for some APIs.

Reply:
What’s the purpose of specifying json in here?